backend/common/db/migrations/versions/2025_12_19_0635-make_user_progress_id_nullable_in_student_instances.py

```
@@ -0,0 +1,126 @@
+        
+        if 'progress_data' not in columns:
+            # SQLite doesn't support JSON type directly, use TEXT
+            with op.batch_alter_table('scene_progress', schema=None) as batch_op:
+                batch_op.add_column(sa.Column('progress_data', sa.Text(), nullable=True))
+    else:
+        # For other databases, try to add column (may fail if exists)
+        try:
+            op.add_column('scene_progress', sa.Column('progress_data', sa.JSON(), nullable=True))
+        except Exception:
+            # Column might already exist, which is fine
+            pass
```

Ruff (0.14.8)

`try`-`except`-`pass` detected, consider logging the exception

(S110)